packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -209,6 +210,24 @@ class UpgradeCommandRunner {
     await runDoctor();
     // Force the welcome message to re-display following the upgrade.
     persistentToolState.setShouldRedisplayWelcomeMessage(true);
+    if (globals.flutterVersion.channel == 'master' || globals.flutterVersion.channel == 'main') {
+      globals.printStatus(
+        '\n'
+        'This channel is intended for Flutter contributors. '
+        'This channel is not as thoroughly tested as the "beta" and "stable" channels. '
+        'We do not recommend using this channel for normal use as it more likely to contain serious regressions.\n'
+        '\n'
+        'If you would like to contribute to Flutter, '
+        'you may need to set up a different development environment. '
+        'For more details, see our contributing guide:\n'
+        '    https://github.com/flutter/flutter/blob/master/CONTRIBUTING.md\n'
+        '\n'
+        'For the most up to date stable version of flutter, consider using the "beta" channel instead. '
+        'The Flutter "beta" channel enjoys all the same thorough testing as the "stable" channel, '
```

Comment:
Is this technically true -- _all_ the same testing? Technically, that can't be quite true, since a stable build has been tested in the beta channel prior to release.

Reply:
how about "automated testing" instead of "thorough testing"?

I mean, right now it's true as written, as we basically don't have any usage of the beta channel. :-)

Reply:
Yeah, that works. Or 'The "beta" channel also enjoys thorough testing, but...'

Reply:
k, changed to "automated testing"